hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/InsertIntoHoodieTableCommand.scala

```diff
@@ -88,17 +88,15 @@ object InsertIntoHoodieTableCommand extends Logging with ProvidesHoodieConfig wi
           extraOptions: Map[String, String] = Map.empty): Boolean = {
     val catalogTable = new HoodieCatalogTable(sparkSession, table)
 
-    var mode = SaveMode.Append
-    var isOverWriteTable = false
-    var isOverWritePartition = false
-    if (overwrite && partitionSpec.isEmpty) {
-      // insert overwrite table
-      mode = SaveMode.Overwrite
-      isOverWriteTable = true
+    val mode = if (overwrite) {
+      SaveMode.Overwrite
```

Comment:
Given the `Overwrite` mode doesn't care abt the old data, do we need to enable `bulk_insert` by default if it's `Overwrite` mode?

Reply:
I think that's a good suggestion. cc @nsivabalan @yihua